### PR TITLE
Feature/impersonate mutation

### PR DESCRIPTION
**Pull request by wender**
**What problem is this solving?**

Adding impersonation capabilities

**How should this be manually tested?**
Log in using a valid user: **wender.lima@gmail.com** `Wender@123`

Direct impersonation using [mutation](https://wender2--sandboxusdev.myvtex.com/_v/private/vtex.storefront-permissions@1.9.1/graphiql/v1?operationName=impersonate&query=mutation%20impersonate(%24userId%3A%20ID)%20%7B%0A%20%20impersonateUser(userId%3A%20%24userId)%20%7B%0A%20%20%20%20status%0A%20%20%20%20message%0A%20%20%7D%0A%7D&variables=%7B%0A%20%20%22userId%22%3A%20%22%22%0A%7D) 

Use a valid `userId` to impersonate, or a empty `userId` to remove impersonation

After executing this mutation, refresh the homepage

**Screenshots or example usage:**
After impersonation:
image

Removed impersonation:
image

[Checking session after refreshing the homepage](https://wender2--sandboxusdev.myvtex.com/api/sessions?items=*)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -75,15 +75,14 @@ The content to this file should be in this format containing the name of your Ap
 | Sales Admin          | `sales-admin`          |
 | Sales Manager        | `sales-manager`        |
 | Sales Representative | `sales-representative` |
-| Admin       | `customer-admin`       |
-| Approver    | `customer-approver`    |
-| Buyer       | `customer-buyer`       |
+| Admin                | `customer-admin`       |
```
Comment (polishq) [inline docs/README.md:78]:
```suggestion
| Customer Admin                | `customer-admin`       |
```

File: docs/README.md [hunk 2]
```diff
@@ -75,15 +75,14 @@ The content to this file should be in this format containing the name of your Ap
...
+| Approver             | `customer-approver`    |
```
Comment (polishq) [inline docs/README.md:79]:
```suggestion
| Customer Approver             | `customer-approver`    |
```

File: docs/README.md [hunk 3]
```diff
@@ -75,15 +75,14 @@ The content to this file should be in this format containing the name of your Ap
...
+| Buyer                | `customer-buyer`       |
```
Comment (polishq) [inline docs/README.md:80]:
```suggestion
| Customer Buyer                | `customer-buyer`       |
```

**Review by polishq — CHANGES_REQUESTED** [review]
@wender Please correct the bug from the SonarCloud analysis.

Also, question - when a user is impersonating, is there a flag added to the session like `isImpersonating: true`? The orgs app will need to be able to check if the current user is currently impersonating someone, like to display a "Stop Impersonation" button. It would also be good to have access to the original user's email and other details, if we wanted to be able to display a message like "You are wender@vtex.com and are currently impersonating arthur@vtex.com".

**Review by thyarles — APPROVED** [review]

**Comment by thyarles** [comment]
Hi there! Any chance to update [axios](https://github.com/vtex-apps/storefront-permissions/security/dependabot/node/yarn.lock/axios/open) and [object-path](https://github.com/vtex-apps/storefront-permissions/security/dependabot/node/yarn.lock/object-path/open) to mitigate the high severity vulnerabilities detected by GitHub?

**Review by thyarles — APPROVED** [review]

**Review by polishq — APPROVED** [review]

**Pull request merged**